pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -449,17 +450,21 @@ func getHashJoins(p *logicalop.LogicalJoin, prop *property.PhysicalProperty) (jo
 		} else {
 			joins = append(joins, getHashJoin(p, prop, 1, false))
 			if forceLeftToBuild || forceRightToBuild {
-				// Do not support specifying the build and probe side for semi join.
-				p.SCtx().GetSessionVars().StmtCtx.SetHintWarning(fmt.Sprintf("We can't use the HASH_JOIN_BUILD or HASH_JOIN_PROBE hint for %s, please check the hint", p.JoinType))
+				p.SCtx().GetSessionVars().StmtCtx.SetHintWarning(fmt.Sprintf(
+					"The HASH_JOIN_BUILD and HASH_JOIN_PROBE hints are not supported for %s with hash join version 1. "+
+						"Please remove these hints",
+					p.JoinType))
 				forceLeftToBuild = false
 				forceRightToBuild = false
 			}
 		}
 	case logicalop.LeftOuterSemiJoin, logicalop.AntiLeftOuterSemiJoin:
 		joins = append(joins, getHashJoin(p, prop, 1, false))
 		if forceLeftToBuild || forceRightToBuild {
-			// Do not support specifying the build and probe side for semi join.
-			p.SCtx().GetSessionVars().StmtCtx.SetHintWarning(fmt.Sprintf("We can't use the HASH_JOIN_BUILD or HASH_JOIN_PROBE hint for %s, please check the hint", p.JoinType))
+			p.SCtx().GetSessionVars().StmtCtx.SetHintWarning(fmt.Sprintf(
+				"HASH_JOIN_BUILD and HASH_JOIN_PROBE hints are not supported for %s because the build side is fixed. "+
+					"The build side will always be the right table. Please remove these hints",
```

Comment:
I am not sure which term is better here. Not sure if we should call it right table or outer table.

Reply:
Or maybe we don't need to mention those details here.

Reply:
Since semi-join always comes from subqueries in the current tidb, I think the "inner table" is easier to understand.
And I think it's also reasonable not to mention this.

Reply:
OK, let's just drop it to reduce the confusion.